### PR TITLE
feat: Add module_call_graph to handle output spec

### PR DESCRIPTION
**Pull request by peri044**
# Description

* This PR adds module_call_graph to preserve the output spec obtained from the graph module. 
* This will let exported program know what the output_spec is during it's construction and ensure the verifier operates properly. 
* This PR should go after https://github.com/pytorch/pytorch/pull/120361 so that the codegen of the partitioned module is populated correctly to add pytree flatten/unflatten nodes accordingly

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/_exporter.py [hunk 1]
```diff
@@ -296,13 +298,25 @@ def create_trt_exp_program(
     # torch.export serialization expects them to be lifted
     gm, trt_graph_signature, state_dict = lift(gm, trt_graph_signature)
 
+    module_call_graph = [
+        ModuleCallEntry(
+            "",
+            ModuleCallSignature(
+                inputs=[],
+                outputs=[],
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/_exporter.py:306]: Are the `ArgumentSpec` specifications not needed here?

**Review by gs-olive — APPROVED** [review]
Overall looks good - added one question for clarification

File: py/torch_tensorrt/dynamo/_exporter.py (hunk 1 above)
Comment (peri044) [reply py/torch_tensorrt/dynamo/_exporter.py:306]: yep not required apparently.
